The organisational challenge of building a Data team: lessons learnt (#255)
M6Web/m6web.github.io: _posts/2020-09-01-organisational-challenge-building-data-team.md

author:
  name:  Cory Chaplin
  avatar: c_chaplin.jpg
  email:
  twitter:  corychaplin
  facebook:
  github: corychaplin
category: Data
tags: [Data,Data Science,Data Engineering,Agile,BigData,Organization]
image:
  feature: posts/data-org-challenge/team.jpg
comments: true
language: en
---

Disclaimer: this is a post I've been meaning to share since the Dailymotion team published [this post](https://medium.com/dailymotion/collaboration-between-data-engineers-data-analysts-and-data-scientists-97c00ab1211f), so it's going way back into our history but updated to our current status. It's also an attempt to bring a counterpart to the blog posts of many web leaders that have written a lot about what has succeeded but rarely detail the difficulties they met and the lessons they learnt.
Here's a very personal (therefore biased) feedback of our story and the lessons we learnt down the road.

At Bedrock we're very busy building the best streaming platform in Europe to help our customers be local streaming champions.

> Within Bedrock (and before that within M6 where our team and initiatives were born), we took the Data move in 2015. From a top management perspective the move was straight forward once the company had understood the strategic importance of Data and the major use cases we could address: staff a team and deliver massive value (since Data was definitely the new oil back in those days). But for the teams we built (now hitting the 40 people), it was another story.

Reviewer: ```suggestion
Within Bedrock (and before that within M6 where our team and initiatives were born), we took the Data move in 2015. From a top management perspective the move was straightforward once the company had understood the strategic importance of Data and the major use cases we could address: staff a team and deliver massive value (since Data was definitely the new oil back in those days). But for the teams we built (now hitting the 40 people), it was another story.
```

## First iteration: the Dream Team

![Dream team](/images/posts/data-org-challenge/dream-team.jpg)

> The first step M6 took was to hire an amazing, PhD qualified, Business focussed, Data Scientist whose mission was to build a Data Science team and explore the company's data to find valuable use cases. A few months later, we added an experienced Product owner (me) to help set the vision, the roadmap and lead our Data projects. And some months later we were joined by a rock star Data/Machine learning Engineer who was to create an engineering team and the data platform we could build upon.

Reviewer: ```suggestion
The first step M6 took was to hire an amazing, PhD qualified, Business focused, Data Scientist whose mission was to build a Data Science team and explore the company's data to find valuable use cases. A few months later, we added an experienced Product owner (me) to help set the vision, the roadmap and lead our Data projects. And some months later we were joined by a rock star Data/Machine learning Engineer who was to create an engineering team and the data platform we could build upon.
```

> This word can take either double or single s, with the single option being highly preferred

From the Data Factory perspective, the new organisation fixed it! Over the past couple of years, we have been building the vision and roadmap within the team with no distinction between the Data Science and Data Engineering roadmap. Our work with the stakeholders feels like we're now walking on our 2 feet, the solutions we imagine for their challenges mix plain data engineering solutions with algorithms and statistics seamlessly, and we're definitely delivering more value. And of course, there is no debate on what goes to production, everything the team does will end up live! At the time I’m writing, the Data Factory has scaled from ~10 people to more than 40 members.

![Split teams](/images/posts/data-org-challenge/teamwork.jpg)

## Organization is one thing, collaboration is another

> Our approach to overcoming our challenges was totally around changing the organization, but I truly believe the solution was also within the evolution of our mindset.

Reviewer: ```suggestion
Our approach to overcome our challenges was totally around changing the organization, but I truly believe the solution was also within the evolution of our mindset.
```

?

> When we officially split the Data Science and Data Engineering teams, we also increased staffing bringing new people into the game. During that round of staffing, we focussed very much on soft skills and mindset, considering that authentic team players would ultimately bring us more value than very strong individuals. Somehow, we moved away from building the dream team and aimed to create a real team. Our idea was that we weren't to deliver much rocket science, just a bunch of features that needed to incorporate some basic data science properly in our pipelines.

Reviewer: ```suggestion
When we officially split the Data Science and Data Engineering teams, we also increased staffing bringing new people into the game. During that round of staffing, we focused very much on soft skills and mindset, considering that authentic team players would ultimately bring us more value than very strong individuals. Somehow, we moved away from building the dream team and aimed to create a real team. Our idea was that we weren't to deliver much rocket science, just a bunch of features that needed to incorporate some basic data science properly in our pipelines.
```

again but it seems to be fine to have 2 `s`, so I don't mind :)
